File: cpp_labs/lab6.cpp
#include <iostream>
#include <math.h>

using namespace std;
float minimum(float local, float min);	//функція визначає найменше значення з двох
int main()
{
	float n, x, y, d, min_d;
	int min_n, i;
	setlocale(LC_ALL, "");
	printf("Дон Василь IС-01\n");
	cout << "Введiть кiлькiсть точок для обчислення: ";
	cin >> n;
	min_n = 0;
	for (i = 1; i <= n; i++)
	{
		cout << "Точка " << i << endl;
		cout << "Введiть х: ";
		cin >> x;
		cout << "Введiть y: ";
		cin >> y;
		d = sqrt(x * x + y * y);
		if (i == 1)
		{
			min_d = d;
		}
		min_d = minimum(d, min_d);
		if (min_d == d)
		{
			min_n = i;
		}
	}
	cout << "Найближча точка до початку координат - точка " << min_n;
	return 0;
}
float minimum(float local, float min)
{
	if (local <= min)
	{
		return local;
	}
	else
	{
		return min;
	}
}

File: py_labs/lab6.py
import math

def perevirka():
    d = math.sqrt(x*x + y*y)
    global min_d, min_n, min_x, min_y
    if i == 1:
        min_d = d
    if d <= min_d:
        min_n = i
        min_x = x
        min_y = y
        min_d = d

print('Дон Василь ІС-01')
n = int(input('Введiть кiлькiсть точок для обчислення: '))
for i in range(1, n+1):
    print('Точка' , i)
    x = float(input('Введіть x:'))
    y = float(input('Введіть y:'))
    perevirka()
print('Найближча точка до початку координат - точка' , min_n , '(' , min_x , ';' , min_y , ')')